main.js

```diff
@@ -145,37 +146,34 @@ function getFormData() {
   formData.begin = beginDate.value;
   formData.end = endDate.value;
   formData.sortBy = sortSelect.value;
-  
-  formData.filters = {};
-  formData.filters.glocation = locationInput.value.trim();
+  formData.glocation = locationInput.value.trim();
 
   const newsDeskFieldset = document.getElementById('newsdesk-fieldset');
-  formData.filters.newsDesks = valuesFromFieldset(newsDeskFieldset);
+  formData.newsDesks = valuesFromFieldset(newsDeskFieldset);
 
   const materialsFieldset = document.getElementById('material-types-fieldset');
-  formData.filters.materialTypes = valuesFromFieldset(materialsFieldset);
+  formData.materialTypes = valuesFromFieldset(materialsFieldset);
   
   return formData;
 }
 
```

Comment:
some of your functions work on global variables some return the value. when we work with classes this will become an important distinction.